### PR TITLE
[SEMI-MODULAR][SOMEWHAT WIP]Pipe tripping

### PR DESCRIPTION
**Pull request by Vizoplez**
## About The Pull Request
Fuck you, Gandalf

## Changelog
:cl:
add: Pipe tripping mechanic
/:cl:

### CODE REVIEW

File: modular_skyrat/master_files/code/datums/traits/skyrat-traits.dm [hunk 1]
```diff
@@ -0,0 +1 @@
+#define TRAIT_NOTRIP "TRAIT_NOTRIP"
```
Comment (Gandalf2k15) [inline modular_skyrat/master_files/code/datums/traits/skyrat-traits.dm]: Denied, no newline

Comment (Vizoplez) [reply modular_skyrat/master_files/code/datums/traits/skyrat-traits.dm]: I dont care

**Comment by Vizoplez** [comment]
so yes the element 'technically' works but fails to actually trip people, p sure this is an invoker issue

**Comment by Vizoplez** [comment]
also, obligatory monkey AI defeating CI. Again

**Comment by Arctaisia** [comment]
Time to line the halls with pipes

**Comment by CameronWoof** [comment]
Why?

**Comment by Swiftfeather** [comment]
Fuck you! >:(

**Comment by Vizoplez** [comment]
@CameronWoof 
image

**Comment by CameronWoof** [comment]
Oh. In that case, I think the problem is that at no point in this code do I see it actually calling a proc to trip anything. It just outputs text.

**Comment by Vizoplez** [comment]
> Oh. In that case, I think the problem is that at no point in this code do I see it actually calling a proc to trip anything. It just outputs text.
image

**Comment by SpookyTheFox** [comment]
> 
> 
> > Oh. In that case, I think the problem is that at no point in this code do I see it actually calling a proc to trip anything. It just outputs text.
> > image

have them like 
Paralyze(5)
add_confusion(50)
dropItemToGround(M.get_active_held_item())
dropItemToGround(M.get_inactive_held_item())
	
when they trip

**Comment by Vizoplez** [comment]
> > > Oh. In that case, I think the problem is that at no point in this code do I see it actually calling a proc to trip anything. It just outputs text.
> > > image
> 
> have them like
> Paralyze(5)
> add_confusion(50)
> dropItemToGround(M.get_active_held_item())
> dropItemToGround(M.get_inactive_held_item())
> 
> when they trip

I'll most likely just make it apply a small stam debuff and a slight slowdown by default, namely for balance concerns

**Comment by MLGTASTICa** [comment]
> so yes the element 'technically' works but fails to actually trip people, p sure this is an invoker issue

use COMSIG_ATOM_CROSSED , that should work with elements , since elements themselves are not on the SPOT of the object they are attached to unless you make them be.

**Comment by KathrinBailey** [comment]
**YOUR INTEGRATION TESTS HAVE BEEN RUMMAGED BY MONKEY BUSINESS**
image
**PAY THE BANANA TAX OR NO COMPILE FOR YOU, OOH OOH OOH!!**

**Comment by Azarak** [comment]
This is a bad idea and bad implementation. Because of how we have piping done on maps, and that maintenance doesnt have any catwalks or such, it's going to be a chore to get around

**Pull request closed without merging**